### DISCUSSION TITLE
libvips cannot read some avif files

### DISCUSSION THREAD

**Discussion by Flash-nDie** [libvips/libvips#3856] [General]
Hello

libvips cannot read some avif files (attached files).

libvips versions and description of images are belows:
```
libvips version: 8.14.0
test1.avif : no orient avif
test2.avif: 90 degree rotate avif
test3.avif: mirror rotate avif 
```
I can see the orient info with heif-info. 

With test1.avif and test3.avif, it works but print out errors 
```
vips heifload test1.avif  t.v (-> works fine)
vips heifsave t.v out.avif (-> works but print errors like under)
heif: heif_writer callback returned a null error text (5.2001)
```

With test2.avif, libvips cannot load image with heifload. I guess libvips doesn't read orient info from avif with libheif. (test3.avif also have orient, but it works with reading widht and height correctly, i think)
```
vips heifload test2.avif t.v
(vips:48385): VIPS-WARNING **: 09:37:48.717: error in tile 0 x 0

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 1

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 2

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 3

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 4

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 5

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 6

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 7

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 8

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 9

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 10

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 11

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 12

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 13

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 14

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 15
test2.avif: bad seek to 84845
heifload: bad image dimensions on decode
```

Can I know the exact reason? and can this be fixed? 

Thank you in advance for your help.

[test.zip](https://github.com/libvips/libvips/files/14303830/test.zip)

**Comment by lovell** [comment]
For images 1 and 3, the "callback returned a null error text" problem when using libheif v1.17.0+ was fixed in libvips v8.15.0 - see https://github.com/libvips/libvips/pull/3724

For image 2, please see https://github.com/libvips/libvips/issues/2551

**Reply by Flash-nDie** [reply]
@lovell 
hello, i tested with libvips 8.15.0. 
As you mentioned, it works with image1 and 3. (no error printed)

For image 2, I see [#2551](https://github.com/libvips/libvips/issues/2551). I think #2551 is about exif tag. When heic file has exif tag and doesnt have irot/imir meta in heic format, exif tag should be ignored. In #2551, the python plugin is introduced.

Anyway, I think image 2 is different case. image 2 has no exif orientation tag and has irot/imir meta in format which is 90 degree rotated. When libvips load image 2 via vips_heifload, it should first detect irot/imir meta, and read the image. But without that, libvips cannot read correctly because of width and height changed, and print below error.
```
vips:48385): VIPS-WARNING **: 09:37:48.717: error in tile 0 x 0

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 1

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 2

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 3

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 4

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 5

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 6

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 7

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 8

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 9

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 10

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 11

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 12

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 13

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 14

(vips:48385): VIPS-WARNING **: 09:37:48.718: error in tile 0 x 15
test2.avif: bad seek to 84845
heifload: bad image dimensions on decode
```
Is there any solution for this ?
